nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/frontend/package-lock.json

```diff
@@ -144,20 +144,20 @@
       "resolved": "https://registry.npmjs.org/d3-brush/-/d3-brush-1.0.4.tgz",
       "integrity": "sha1-AMLyOAGfJPbAoZSibUGhUw/+e8Q=",
       "requires": {
-        "d3-dispatch": "1.0.3",
-        "d3-drag": "1.2.1",
-        "d3-interpolate": "1.1.6",
-        "d3-selection": "1.3.0",
-        "d3-transition": "1.1.1"
+        "d3-dispatch": "1",
```

Comment:
This change may appear strange but what it is saying is that "d3-brush" version 1.0.4 requires any minor or patch release of "d3-dispatch" version 1. This however does not change the version of "d3-dispatch" that is installed. That is defined here: https://github.com/apache/nifi/pull/4449/files#diff-9a4616626c8b30875e090d2d589ce665R173.

Reply:
I can confirm this is how their new versioning approach works and that `requires` is just used for version mismatch warnings. The top-level dependencies determines what is installed.

Reply:
@scottyaslan @sardell Thanks for the analysis here. If I've understood these comments correctly, they explain the behavior for a direct dependency of NiFi. They even touch on what happens is a conflict in versioned between a direct dependency of NiFi that is also a transitive dependency. 

But what is the behavior for something that is only a transitive dependency and there is no direct dependency to compare against? If that `requires` section does not contain a specific version, what ensures that when newer versions become available they aren't used? Looking to confirm that we can claim that we have repeatable builds.

Reply:
@scottyaslan @sardell I think I misunderstood the top-level dependency comment. It looks like all dependencies (transitive or not) will be 'top-level' in the package-lock.json. These will have explicit versions and will ensure repeatable builds assuming we use it in our builds.